### PR TITLE
Add delay time for sync Magnum

### PR DESCRIPTION
**Pull request by yakiduck**
Add flag: -cr-remove-delay-interval
Add delay time after cr has created before Magnum writes to the database

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -64,7 +64,9 @@ func main() {
 	pollingPeriod := flag.Duration("polling-period", 13*time.Second, "The polling loop period.")
 	openstackPeriod := flag.Duration("openstack-period", 23*time.Second, "The polling loop period.")
 	tagPeriod := flag.Duration("tag-period", 900*time.Second, "The polling loop period.")
-	syncdu := flag.Duration("sync-period", 30*time.Second, "controller manager sync resource time duration")
+	syncdu := flag.Duration("sync-period", 30*time.Second, "Controller manager sync resource time duration")
+	clusterNotFoundDelay := flag.Duration("cluster-not-found-delay-seconds", 30*time.Minute,
```
Comment (yylt) [inline main.go]: `xxx-period` is more reasonable

File: pkg/api/v1/cluster_types.go [hunk 2]
```diff
@@ -92,6 +93,8 @@ type EksSpec struct {
 	EksHealthReasons  map[string]string `json:"eks_health_reasons,omitempty"`
 
 	Hadsync bool `json:"-"`
+	// ClusterNotFoundTimestamp record first time cluster is not found (404)
+	ClusterNotFoundTimestamp time.Time `json:"-"`
```
Comment (yylt) [inline pkg/api/v1/cluster_types.go:97]: should be in `ClusterStatus`

File: pkg/controllers/cluster.go [hunk 3]
```diff
@@ -135,9 +137,24 @@ func (c *Operate) mgFilter(page pagination.Page) {
 				info, err := clusters.Get(cli, id).Extract()
 				if err != nil {
 					if _, ok := err.(gophercloud.ErrDefault404); ok {
+						oldStatus := c.magnums[id].EksStatus
+						if oldStatus == "" {
+							timeNow := time.Now().UTC()
```
Comment (yylt) [inline pkg/controllers/cluster.go]: `UTC() ` not necessary

File: pkg/controllers/cluster.go [hunk 4]
```diff
@@ -135,9 +137,24 @@ func (c *Operate) mgFilter(page pagination.Page) {
...
+							timestamp := c.magnums[id].ClusterNotFoundTimestamp
+							if timestamp.IsZero() {
+								c.magnums[id].ClusterNotFoundTimestamp = timeNow
+							}
+							timeSub := timeNow.Sub(timestamp)
```
Comment (yylt) [inline pkg/controllers/cluster.go]: maybe use `time.Add(duration).After(oldtime)` more reasonable

File: main.go (hunk 1 above)
Comment (yylt) [reply main.go]: should compare with MinDuretion（5min）

Comment (yakiduck) [reply main.go]: just like InitialDelaySeconds, it is not a period

File: pkg/api/v1/cluster_types.go (hunk 2 above)
Comment (yakiduck) [reply pkg/api/v1/cluster_types.go:97]: It is not a generic property. it is only for eks cluster.

File: pkg/controllers/cluster.go (hunk 4 above)
Comment (yakiduck) [reply pkg/controllers/cluster.go]: Done

File: pkg/controllers/cluster.go (hunk 3 above)
Comment (yakiduck) [reply pkg/controllers/cluster.go]: Done

**Pull request merged**